### PR TITLE
added buyReservedMine reaction

### LINKED ISSUE #106: Implement BuyReservedMine reaction
We need to implement a similar reaction to issue #62, but this time, it should allow players to buy a card they have reserved earlier.

We decided to split the action of buying mines into two separate reactions. This is because if a user buys a mine from the table (not reserved), the server needs to send information about the new card that was drawn from the deck. However, if a user buys a card from their reserved set, there is no need to draw a new card or send this information to the client. Instead of sending `null` and adding more if statements, we can split this at the class and type level.

### PR DESCRIPTION
**Pull request by VEXI19**
Also created isCardReserved in a User class

### CODE REVIEW

File: src/main/java/com/github/splendor_mobile_game/App.java
Comment (kestivvi) [inline src/main/java/com/github/splendor_mobile_game/App.java:1]: Please, do not format already pretty code, that has been written by someone else if this pull request doens't serve this purpose. Please, revert it.

**Review by kestivvi — CHANGES_REQUESTED** [review]

**Comment by kestivvi** [comment]
Nicely written class ✨✨

I haven't test that yet. Also please, turn of your code formatter. It's destroying some pieces of code xDDDDDDD

**Review by Vertonowsky — DISMISSED** [review]

**Comment by kestivvi** [comment]
Now you have to resolve merge conflicts as well 🗿

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/BuyReservedMine.java [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+        DataDTO dataDTO = (DataDTO) userMessage.getData();
+
+        try {
+            validateData(dataDTO, database);
+
+            User buyer = database.getUser(dataDTO.userDTO.uuid);
+            Card boughtCard = database.getCard(dataDTO.cardDTO.uuid);
+            Room room = database.getRoomWithUser(buyer.getUuid());
+
+            buyer.buyCard(boughtCard);
```
Comment (kestivvi) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/BuyReservedMine.java:168]: You forgot about removing this card from reserved cards.

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/BuyReservedMine.java [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+        Game game = room.getGame();
+
+        // Check if user is in game
+        if (game == null) {
+            throw new RoomInGameException("The room is not in game state");
+        }
+
+        // Check if it is user's turn
+        if (game.getCurrentPlayer() != player) {
+            throw new UserTurnException("It is not a user's turn");
```
Comment (kestivvi) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/BuyReservedMine.java:233]: If it is user's turn, you should also check if this user has did some action. Use `hasPerformedAction` field that is in `User`.

**Review by kestivvi — CHANGES_REQUESTED** [review]
I hadn't been meticulous then and now I've found things to be improved. Sry about that 😥

File: src/main/java/com/github/splendor_mobile_game/game/model/User.java [hunk 3]
```diff
@@ -164,6 +164,10 @@ public boolean isCardReserved(Card card) {
         return test;
     }
 
+    public void removeCardFromResrved(Card card) {
```
Comment (kestivvi) [inline src/main/java/com/github/splendor_mobile_game/game/model/User.java]: Is this a typo?

**Review by kestivvi — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/splendor_mobile_game/game/model/User.java (hunk 3 above)
Comment (VEXI19) [reply src/main/java/com/github/splendor_mobile_game/game/model/User.java]: in did, it is

Comment (VEXI19) [reply src/main/java/com/github/splendor_mobile_game/game/model/User.java]: well spotted

**Review by kestivvi — DISMISSED** [review]
Great work!!! 🎉🎊🎊👌

**Review by kestivvi — APPROVED** [review]
👌🎊🎉🎈

**Review by Vertonowsky — APPROVED** [review]

**Pull request merged**